File: String/Python/_1309_DecryptStringfromAlphabettoIntegerMapping.py
class Solution:
    # Iteration
    # Time Complexity: BigO(N)
    # Space Complexity: BigO(N)
    def freqAlphabets(self, s: str) -> str:
        res = []
        tmp = ""
        for x in reversed(s):
            if len(tmp) == 1 or x == "#":
                tmp = x + tmp
                continue

            tmp = x + tmp
            tmp = tmp.replace("#", "")
            res.insert(0, chr(96 + int(tmp)))
            tmp = ""

        return "".join(res)


ins = Solution()

print(ins.freqAlphabets("10#11#12"))
print(ins.freqAlphabets("1326#"))

File: String/TypeScript/_1309_DecryptStringfromAlphabettoIntegerMapping.ts
function freqAlphabets(s: string): string {
  let res: string[] = [];
  for (let i = s.length - 1; i >= 0; i--) {
    if (s[i] == "#") {
      let chr = s[i - 2] + s[i - 1] + s[i];
      res.unshift(String.fromCharCode(96 + parseInt(chr, 10)));
      i -= 2;
    } else {
      res.unshift(String.fromCharCode(96 + parseInt(s[i], 10)));
    }
  }
  return res.join("");
}

console.log(freqAlphabets("10#11#12"));
console.log(freqAlphabets("1326#"));
